storage/boltdb/client_test.go

```diff
@@ -140,3 +149,155 @@ func BenchmarkSuiteLong(b *testing.B) {
 	}
 	testsuite.BenchmarkPathOperationsInLargeDb(b, longStore)
 }
+
+func BenchmarkClientWrite(b *testing.B) {
+	// setup db
+	ctx := testcontext.New(b)
+	defer ctx.Cleanup()
+	dbfile := ctx.File("testbolt.db")
+	dbs, err := NewShared(dbfile, "kbuckets", "nodes")
+	if err != nil {
+		fmt.Printf("failed to create db: %v\n", err)
+	}
+	defer func() {
+		if err := dbs[0].Close(); err != nil {
+			fmt.Printf("failed to close db: %v\n", err)
+		}
+		if err := dbs[1].Close(); err != nil {
+			fmt.Printf("failed to close db: %v\n", err)
+		}
+	}()
+	kdb := dbs[0]
+
+	// benchmark test: execute 1000 Put operations where each call to `PutAndCommit` does the following:
+	// 1) create a BoltDB transaction (tx), 2) execute the db operation, 3) commit the tx which writes it to disk.
+	for n := 0; n < b.N; n++ {
+		for i := 0; i < 1000; i++ {
+			key := storage.Key(fmt.Sprintf("testkey%d", i))
+			value := storage.Value("testvalue")
+
+			err := kdb.PutAndCommit(key, value)
```

Comment:
BenchmarkClientWrite() and BenchmarkClientNoSyncWrite() would be more meaningful comparisons if they also used the same WaitGroup/goroutine pattern as BenchmarkClientBatchWrite() and BenchmarkClientBatchNoSyncWrite(). It might even be worth pulling out the commonalities between them all into a single runner function that takes a callback argument for doing the actual appropriate operation.

Reply:
hm ok let me give that a shot